Source/Core/VideoCommon/RenderBase.cpp

```diff
@@ -1481,6 +1485,87 @@ void Renderer::Swap(u32 xfb_addr, u32 fb_width, u32 fb_stride, u32 fb_height, u6
   }
 }
 
+void Renderer::UISwap()
+{
+  if (m_xfb_entry)
```

Comment:
No reason to skip rendering the whole UI when you don't have an XFB. 
You should be able to skip just the `RenderXFBToScreen` and imgui will render over a blank screen.